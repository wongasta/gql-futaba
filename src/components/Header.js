//@flow
import * as React from "react";
import { Link } from "react-router-dom";
import styles from "./Header.module.css";
import Logo from "../assets/logo.png";

export default function Header():React.Node{
  return (
    <header className={styles.header}>
      <Link to={"/posts"}><img src={Logo} className={styles.logo} alt={"GQL"} /></Link>
      <h1 className={styles.text}>/gql/ - GraphQL</h1>
    </header>
  )
}