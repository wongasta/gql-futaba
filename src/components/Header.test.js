import React from 'react';
import { mount } from "enzyme";
import { MemoryRouter } from 'react-router'
import Header from './Header';
import toJson from "enzyme-to-json";

describe("Header", ()=>{
  test("Construction Snapshot", ()=>{
    const headerWrapper = mount(<MemoryRouter><Header /></MemoryRouter>);
    const headerWrapperInstance = headerWrapper.find(Header);
    expect(toJson(headerWrapperInstance)).toMatchSnapshot();
  })
})