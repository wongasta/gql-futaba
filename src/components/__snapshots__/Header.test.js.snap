// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Construction Snapshot 1`] = `
<Header>
  <header
    className="header"
  >
    <Link
      to="/posts"
    >
      <LinkAnchor
        href="/posts"
        navigate={[Function]}
      >
        <a
          href="/posts"
          onClick={[Function]}
        >
          <img
            alt="GQL"
            className="logo"
            src="logo.png"
          />
        </a>
      </LinkAnchor>
    </Link>
    <h1
      className="text"
    >
      /gql/ - GraphQL
    </h1>
  </header>
</Header>
`;
